blogposts/inside-rust/02-deallocate-stack-frames.md

```diff
@@ -0,0 +1,124 @@
+"stack-deallocating" functions.
+
+## Our constraints
+
+Taking a step back, we have two mandatory constraints on our design:
+
+* There must be sound way to call `libc` functions that may `pthread_cancel`.
+* There must be a sound way for Rust code to invoke C code that may `longjmp`
+  over Rust frames.
+
+In addition, we would like to adhere to several design principles:
+
+* The specified behavior can't be target-platform-specific; in other words, our
+  specification of Rust's interaction with `longjmp` should not depend on
+  whether `longjmp` deallocates frames or initiates a forced-unwind.
+  Optimizations, however, *can* be target-platform-specific.
+* There should be no difference in the specified behavior of frame-deallocation
+  performed by `longjmp` versus that performed by `pthread_cancel`.
+* We will only permit canceling POFs.
+
+## POFs and stack-deallocating functions
+
+The `"C unwind"` RFC introduced a new concept designed to help us deal with
+force-unwinding or stack-deallocating functions: the POF, or "Plain Old Frame".
+<!-- TODO: link to section, short explanation here -->
+
+From the definition, it should be clear that it is dangerous to call a
+stack-deallocating function in a context that could destroy a non-POF stack
+frame. A simple specification for Rust's interaction with stack-deallocating
+functions, then, could be that they are safe to call as long as only POFs are
+deallocated. This would make Rust's guarantees for `longjmp` essentially the
+same as C++'s.
+
+For now, however, we are considering POFs to be "necessary but not sufficient."
+We believe that a more restrictive specification may provide the following
+advantages:
+
+* more opportunities for helpful compiler warnings or errors to prevent misuse
+  of stack-deallocation functions
+* semantic tracatbility: we can make reliance on stack-frame-deallocation
+  visible for all functions involved
+* increased optimization potential when cleanup is "guaranteed" (i.e., the
+  compiler may turn a POF into a non-POF if it knows that this is safe and that
+  the newly inserted cleanup operation is necessary for an optimization)
+
+## Annotating POFs
+
+Our current plan is to introduce a new annotation for frames that are intended
+to be safe to deallocate. These functions, of course, must be POFs, and they
+also cannot invoke any functions without this annotation. This makes
+stack-deallocation "transitive", just like `async`: functions without this
+annotation either must not invoke any annotated functions or must guarantee
+that they will cause the stack-deallocation to terminate (for instance, a
+non-POF, non-annotated function may call `setjmp`).
+<!-- TODO improve explanation -->
+
+The name of the annotation should be based on the terminology used to refer to
+functions that are safe to deallocate. Because this terminology is not
+finalized, we do not yet have a name for the annotation.
```

Comment:
I think a few examples would make this much more concrete. I'd give this a "dummy name" like `#[longjmp_like]` or something.